docs/swagger/spec.yml

```diff
@@ -184,6 +186,259 @@ paths:
+          description: Paginated list of images associated with this Layer
+          schema:
+            $ref: '#/definitions/LayerImagePaginated'
+    post:
+      summary: Create a new association between an image and this layer
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/uuid'
+        - name: image
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/LayerImage'
+      responses:
+        202:
+          description: Layer image details; thumbnails and boundary may be in-progress
+          schema:
+            $ref: '#/definitions/LayerImage'
+        200:
+          description: Layer image details; the image already existed
+          schema:
+            $ref: '#/definitions/LayerImage'
+  /layer-images/:
+    get:
+      summary: Paginated list of layer images
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/orderingBase'
+        - $ref: '#/parameters/pageSize'
+        - $ref: '#/parameters/page'
+      responses:
+        200:
+          description: Paginated list of images
+          schema:
+            $ref: '#/definitions/LayerImagePaginated'
+    post:
+      summary: Create a new image not associated with any Layer
+      tags:
+        - Layers
+      parameters:
+        - name: image
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/LayerImage'
+      responses:
+        202:
+          description: Layer image details; thumbnails and boundary in-progress
+          schema:
+            $ref: '#/definitions/LayerImage'
+  /layer-images/{uuid}:
+    get:
+      summary: Details of a LayerImage
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/uuid'
+      responses:
+        200:
+          description: Image details (All task-populated fields have final values)
+          schema:
+            $ref: '#/definitions/LayerImage'
+        202:
+          description: Image details (some task-populated fields are in progress)
+          schema:
+            $ref: '#/definitions/LayerImage'
+        404:
+          description: Image UUID was not found
+          schema:
+            $ref: '#/definitions/Error'
+    put:
+      summary: Update a LayerImage
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/uuid'
+        - name: image
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/LayerImage'
+      responses:
+        202:
+          description: Update successful, further processing required
+          schema:
+            $ref: '#/definitions/LayerImage'
+        200:
+          description: Update successful, no further processing required
+          schema:
+            $ref: '#/definitions/LayerImage'
+        404:
+          description: Image UUID was not found
+          schema:
+            $ref: '#/definitions/Error'
+  /thumbnails/:
```

Comment:
There are not nested endpoints for thumbnails or boundaries (e.g. `/layers/<uuid>/thumbnails/`) because each layer and layer image will always have a fixed set of thumbnails and boundaries so they are included directly.